frame/merkle-mountain-range/src/lib.rs

```diff
@@ -318,38 +318,75 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			.saturating_add(leaf_index.saturated_into())
 	}
 
-	/// Generate a MMR proof for the given `leaf_indices`.
+	/// Convert a `block_num` into a leaf index.
+	fn block_num_to_leaf_index(block_num: T::BlockNumber) -> Result<LeafIndex, primitives::Error>
+	where
+		T: frame_system::Config,
+		T::BlockNumber: From<LeafIndex>,
```

Comment:
Now that we don't make any assumptions about the `BlockNumber` and `LeafIndex` types anymore, there is an error because in the `kitchensink` runtime the `BlockNumber` is a `u32`, while the `LeafIndex` is a `u64`. Should I update the `BlockNumber` type inside the `kitchensink` runtime?

Reply:
cc @acatangiu

Reply:
No, the `BlockNumber` substrate trait has a bound on `AtLeast32Bit`, but no guarantee above that (64bit). So this pallet should do the same. I believe you had a `saturating_*` solution earlier that worked fine.